File: hw/tipan/u3/hw24AbstractFactory/AbstractFactoryPython/model/gui_factory.py
from abc import ABC, abstractmethod

class GUIFactory(ABC):
    @staticmethod
    def get_factory():
        sys = 0  # 0 para Windows, 1 para Linux
        
        if sys == 0:
            from model.win_factory import WinFactory
            return WinFactory()
        else:
            from model.linux_factory import LinuxFactory
            return LinuxFactory()

    @abstractmethod
    def create_button(self):
        pass

    @abstractmethod
    def create_menu(self):
        pass

File: hw/tipan/u3/hw24AbstractFactory/AbstractFactoryJavaScript/model/guiFactory.js
import { WinFactory } from './winFactory.js';
import { LinuxFactory } from './linuxFactory.js';

export class GUIFactory {
    static getFactory() {
        // Cambia a 0 para Windows, 1 para Linux
        const sys = 0;
        
        if (sys === 0) {
            return new WinFactory();
        } else {
            return new LinuxFactory();
        }
    }

    createButton() {
        throw new Error("Method 'createButton()' must be implemented");
    }

    createMenu() {
        throw new Error("Method 'createMenu()' must be implemented");
    }
}